snap/hooks/install

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+set -ex
+
+# Copy Spack Config
+# - $SNAP_COMMON/config.yaml       <-- typically /var/snap/spack/common/config.yaml
+# env; cp $SNAP/config.yaml $SNAP_COMMON/config.yaml
+SPACK=/opt/$SNAP_NAME
+
+ if [ ! -d "$SPACK" ]; then
+    mkdir $SPACK
+    chmod 777 $SPACK
+fi
+
+# Copy over Spack configs
+cp $SNAP/etc/spack/defaults/config.yaml $SPACK/config.yaml
+cp $SNAP/etc/spack/defaults/modules.yaml $SPACK/modules.yaml
+
+# Update to use $SNAP_USER_COMMON
+# TODO: Make it so $SNAP_USER_COMMON is /home/user/snap/spack/common instead of current /root/snap/spack/common
+sed -i 's@\$spack@'"\${SNAP_USER_COMMON}"'@g' $SPACK/config.yaml
+sed -i 's@\$spack@'"\${SNAP_USER_COMMON}"'@g' $SPACK/modules.yaml
```

Comment:
Do you still want to be performing this substitution here? We should probably make sure the path is pointing to _/opt/spack_.

Reply:
Hmm, I might wait a day or two till the discussion on the spec around this closes first. If there's no movement on the spec for a few days then I think I could move it to _/opt/spack_ till/if we change our minds later.

Reply:
Switched this over to /opt/spack per paired-programming session